### PR TITLE
EditableHeading and Checkbox typescript migration

### PR DESCRIPTION
**Pull request by orrgottlieb**

### CODE REVIEW

File: src/components/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -9,11 +8,31 @@ import { backwardCompatibilityForProperties } from "../../helpers/backwardCompat
...
+  disabled?: boolean;
+  defaultChecked?: boolean;
+  value?: string;
+  name?: string;
+  id?: string;
+}
 
 const BASE_CLASS_NAME = "monday-style-checkbox";
 
 // TODO TS-migration
```
Comment (SergeyRoyt) [inline src/components/Checkbox/Checkbox.tsx:34]: can be removed now :)

File: src/components/EditableHeading/EditableHeading.tsx [hunk 2]
```diff
@@ -1,34 +1,67 @@
...
+  types?: typeof HeadingTypes;
+} = props => {
   const {
-    id,
+    id = "",
     className,
-    inputClassName,
-    dataTestId,
+    inputClassName = "",
+    dataTestId = "",
```
Comment (SergeyRoyt) [inline src/components/EditableHeading/EditableHeading.tsx:45]: maybe to change to `"data-testid": dataTestId` as this is our new convention?

Comment (SergeyRoyt) [inline src/components/EditableHeading/EditableHeading.tsx:45]: maybe to change to `"data-testid": dataTestId` as this is our new convention?

File: src/components/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -9,11 +8,31 @@ import { backwardCompatibilityForProperties } from "../../helpers/backwardCompat
...
-export const Checkbox = forwardRef(
+export const Checkbox: VibeComponent<CheckBoxProps, unknown> = forwardRef(
```
Comment (SergeyRoyt) [inline src/components/Checkbox/Checkbox.tsx:35]: `unknown` can be removed

File: src/components/EditableHeading/EditableHeading.tsx [hunk 4]
```diff
@@ -1,34 +1,67 @@
...
+  tooltip?: string;
+  highlightTerm?: string;
+  insetFocus?: boolean;
+  contentRenderer?: React.FC;
+}
+
+const EditableHeading: React.FC<EditableHeadingProps> & {
+  sizes?: typeof HeadingSizes;
+  types?: typeof HeadingTypes;
+} = props => {
```
Comment (SergeyRoyt) [inline src/components/EditableHeading/EditableHeading.tsx:40]: maybe it'd better to deconstruct props here, as we always do?

**Review by SergeyRoyt — APPROVED** [review]
Approved with a few notes

File: src/components/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (orrgottlieb) [reply src/components/Checkbox/Checkbox.tsx:34]: 😎

File: src/components/EditableHeading/EditableHeading.tsx (hunk 2 above)
Comment (orrgottlieb) [reply src/components/EditableHeading/EditableHeading.tsx:45]: 👍

File: src/components/Checkbox/Checkbox.tsx (hunk 3 above)
Comment (orrgottlieb) [reply src/components/Checkbox/Checkbox.tsx:35]: done

File: src/components/EditableHeading/EditableHeading.tsx [hunk 5]
```diff
@@ -1,34 +1,67 @@
...
+export interface EditableHeadingProps extends EditableInputProps, HeadingProps {
+  displayPlaceholderInTextMode?: boolean;
+  suggestEditOnHover?: boolean;
+  autoSize?: boolean;
+  inputAriaLabel?: string;
+  placeholder?: string;
+  errorClass?: string;
+  ignoreBlurClass?: string;
+  id?: string;
+  className?: string;
```
[the remark this answers is not in the document]
Comment (orrgottlieb) [reply src/components/EditableHeading/EditableHeading.tsx:21]: good catch removeing

File: src/components/EditableHeading/EditableHeading.tsx (hunk 4 above)
Comment (orrgottlieb) [reply src/components/EditableHeading/EditableHeading.tsx:40]: @SergeyRoyt 
we have tons of overrides with the same name
this component requires heavy refactoring - not in this PR though

**Pull request merged**